ui2/src/common/TableTransfer.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { Transfer } from 'antd';
+import _ from 'lodash';
+import { AppTable } from './Table';
+import { Link } from 'react-router-dom';
+import { AppButton } from '../common/Button';
+import { tableColumns } from '../utils/tableColumns';
+import './TableTransfer.css';
+
+export const TableTransfer = ({
+  data,
+  searchColumns,
+  rightColTitles,
+  leftColTitles,
+  selectGroup,
+  predefinedGroups,
+  linkTo,
+  addSearch,
+  ...props
+}) => {
+  const [targetKeys, setTargetKeys] = useState([]);
+  const [showSearch] = useState(!!searchColumns);
+  const formattedData = data.map(dataItem => ({ ...dataItem, key: dataItem.id.toString() }));
+
+  const handleSearch = (inputValue, item) => {
+    return searchColumns.map(searchCol => item[searchCol] && item[searchCol].indexOf(inputValue) !== -1).includes(true);
+  };
```

Comment:
what is thisss

Reply:
OMG

Reply:
ok so you return true if the input is in one of the columns 
so:  
  const handleSearch = (inputValue, item) => searchColumns.includes(searchCol => item[searchCol].contains(inputValue))
  };

Reply:
it doesn't work, I'm sorry that it is so ugly.. will fix it tomorrow.